### PR TITLE
Moving some functionality to MakerOrders contract

### PR DESCRIPTION
**Pull request by amitkot**
The responsibility for keeping track of each makers free orders is
moved to a new layer that the PermissionLessReserve now extends.

### CODE REVIEW

**Comment by amitkot** [comment]
@ilanDoron

File: contracts/MakerOrders.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    struct FreeOrders {
+        uint32 firstOrderId;
+        uint32 numOrders; //max is 256
+        uint8 maxOrdersReached;
+        uint256 takenBitmap;
+    }
+
+    //each maker will have orders that will be reused.
+    mapping(address => FreeOrders) public makerOrdersSell;
+    mapping(address => FreeOrders) public makerOrdersBuy;
```
Comment (ilanDoron) [inline contracts/MakerOrders.sol]: thought the mapping should be in extending contract. its not used here anyway. and this layer don't have to be aware of sell buy diff.

thats what we talked...

File: contracts/MakerOrders.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    function takeOrderId(FreeOrders storage freeOrders)
+        internal
+        onlyAdmin
```
Comment (ilanDoron) [inline contracts/MakerOrders.sol]: since internal and extended. don't need only Admin no more...

File: contracts/MakerOrders.sol [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                freeOrders.takenBitmap = orderBitmap | bitPointer;
+                return(uint32(uint(freeOrders.firstOrderId) + i));
+            }
+
+            bitPointer *= 2;
+        }
+
+        freeOrders.maxOrdersReached = 1;
+
+        require(false);
```
Comment (ilanDoron) [inline contracts/MakerOrders.sol]: should decide if to use maxOrdersReached and avoid reverting.
or revert.

if we revert. maxOrdersReached will not be set anyway.

File: contracts/MakerOrders.sol [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    /// @dev mark order as free to use.
+    function releaseOrderId(FreeOrders storage freeOrders, uint32 orderId)
+        internal
+        onlyAdmin
```
Comment (ilanDoron) [inline contracts/MakerOrders.sol]: again. since internal don't need onlyAdmin

File: contracts/MakerOrders.sol [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        freeOrders.takenBitmap &= bitNegation;
+    }
+
+    function allocateOrders(
+        FreeOrders storage freeOrders,
+        uint32 firstAllocatedId,
+        uint32 howMany
+    )
+        internal
+        onlyAdmin
```
Comment (ilanDoron) [inline contracts/MakerOrders.sol]: remove onlyAdmin

File: contracts/Orders.sol [hunk 6]
```diff
@@ -260,73 +260,12 @@ contract Orders is Withdrawable, Utils2 {
         );
     }
 
-    function getNextOrder(uint32 orderId) public view returns(uint32, bool isLast) {
+    function getNextOrder(uint32 orderId)
+        public
+        view
+        returns(uint32, bool isLast)
+    {
```
Comment (ilanDoron) [inline contracts/Orders.sol]: I prefer the one line syntax if possible
Don't u?

File: contracts/MakerOrders.sol (hunk 2 above)
Comment (amitkot) [reply contracts/MakerOrders.sol]: Done.

File: contracts/MakerOrders.sol (hunk 4 above)
Comment (amitkot) [reply contracts/MakerOrders.sol]: Done.

File: contracts/MakerOrders.sol (hunk 5 above)
Comment (amitkot) [reply contracts/MakerOrders.sol]: Done.

File: contracts/Orders.sol (hunk 6 above)
Comment (amitkot) [reply contracts/Orders.sol]: Following the Solidity style guide, if it exceeds the line length we should break it.

File: contracts/MakerOrders.sol (hunk 1 above)
Comment (amitkot) [reply contracts/MakerOrders.sol]: Done.

File: contracts/MakerOrders.sol (hunk 3 above)
Comment (amitkot) [reply contracts/MakerOrders.sol]: Let's discuss this.

**Pull request merged**